cmd/global-query/pkg/distributed/querier.go

```diff
@@ -2,101 +2,36 @@ package distributed
 
 import (
 	"context"
-	"fmt"
-	"os"
-	"path/filepath"
 
 	"github.com/els0r/goProbe/cmd/global-query/pkg/hosts"
-	"github.com/els0r/goProbe/pkg/api/goprobe/client"
 	"github.com/els0r/goProbe/pkg/query"
 	"github.com/els0r/goProbe/pkg/results"
-	"gopkg.in/yaml.v3"
 )
 
 // QuerierType denotes the type of the querier instance
 type QuerierType string
 
 const (
-
-	// APIClientQuerierType provides the name for the API-based querier
+	// APIClientQuerierType provides the name for the goProbe API-based querier
 	APIClientQuerierType QuerierType = "api"
 )
 
 // Querier provides a general interface for all query executors
 type Querier interface {
-
-	// CreateQueryWorkload prepares and executes the workload required to perform the query
-	CreateQueryWorkload(ctx context.Context, host string, args *query.Args) (*QueryWorkload, error)
+	// Query runs the distributed query on the provided hosts and returns a channel from
+	// which the results can be read. It is the responsibility of the implementing type
+	// to close the channel.
+	// This may become a requirement through the interface definitions in future versions
+	Query(ctx context.Context, hosts hosts.Hosts, args *query.Args) <-chan *results.DistributedResult
```

Comment:
Why `DistributedResult`? This isn't limited to that case, is it? Take a local raw query via `goQuery` for example, or a non-distributed API query to a single instance of `goProbe` (the latter can probably be subsumed, but still the naming might be misleading, no)?

Reply:
Heya. I was torn between adding the `Hostname` and the `Error` to `results.Result`, just because this is for distributed queries only. That concept will only be used by `global-query` btw.

`goQuery` will use the `globalquery.Client` to run the query. That one doesn't use `distributed.Result` at all, so we are good.

Reply:
Mh, but then I'm confused. Shouldn't all these use the same interface (after all I have to implement the raw query for local goQuery as well / in particular). Or maybe I'm missing something here. :thinking:

Reply:
K. Let's keep it crisp and add those two fields to the `Results` structure. Will have to adapt OpenAPI then, but that's life: not fair and too short...

Reply:
Sounds good, thx. I'll be happy (ok, that's an overstatement, but you get the sentiment) to take care of the OpenAPI stuff if you like me to. Makes life a little more fair. :wink:

Reply:
All good dude. The way it's set up, you just change it in one place.